test/connect.js

```diff
@@ -350,16 +350,16 @@ test('reject second CONNECT Packet sent while first CONNECT still in preConnect
   }
 
   (async () => {
-    try {
-      await Promise.all([msg(s, 100, packet1), msg(s, 200, packet2)])
-      setImmediate(() => {
-        broker.close()
-        t.end()
-      })
-    } catch (error) {
+    await Promise.all([msg(s, 100, packet1), msg(s, 200, packet2)])
+    setImmediate(() => {
```

Comment:
I would make this a promise and use await.

Reply:
```
  function finish () {
    return new Promise((resolve, reject) => {
      broker.close()
      resolve(t.end)
    })
  }
  (async () => {
    await Promise.all([msg(s, 100, packet1), msg(s, 200, packet2)])
    await finish()
  })().catch(
    (error) => {
      t.fail(error)
    }
  )
```
Is the above what you mean? :)

Reply:
or should it be like this?
```
  (async () => {
    await Promise.all([msg(s, 100, packet1), msg(s, 200, packet2)])
  })().catch(
    (error) => {
      t.fail(error)
    }
  ).finally(finish)
```

Reply:
I mean

```js
const immediate = promisify(setImmediate)
```

Reply:
awesome!. I haven't seen we could promisify `setImmediate`. 
Like this?
```
  const immediate = util.promisify(setImmediate)

  ;(async () => {
    await Promise.all([msg(s, 100, packet1), msg(s, 200, packet2)])
    await immediate().then(() => {
      broker.close()
      t.end()
    })
  })().catch(
    (error) => {
      t.fail(error)
    }
  )
```

Reply:
> I would make this a promise and use await.

May we know what the differences are?